### PR TITLE
[Slider] Use MDCPalettes for default thumb track color

### PR DESCRIPTION
**Pull request by sigito**
Partially addresses #1221 and #1222.

### CODE REVIEW

**Review by randallli — APPROVED** [review]

**Comment by randallli** [comment]
I would like @jmdetloff to review as well.

File: components/Slider/src/MDCSlider.m [hunk 1]
```diff
@@ -329,10 +323,6 @@ - (void)thumbTrackTouchCanceled:(__unused MDCThumbTrack *)thumbTrack {
   [self sendActionsForControlEvents:UIControlEventTouchCancel];
 }
 
-+ (UIColor *)defaultColor {
```
Comment (randallli) [inline components/Slider/src/MDCSlider.m:332]: just an fyi: we may have set this up for subclass use. See as we are discouraging that im ok with removing it.

Comment (sigito) [reply components/Slider/src/MDCSlider.m:332]: Sure, but I didn't see any uses of this method, so removed. It is also not in the [documentation](https://github.com/material-components/material-components-ios/blob/develop/components/Slider/README.md).

**Comment by ianegordon** [comment]
Hi @sigito Can you rebase on top of origin/develop to eliminate the conflict and force push?
Once CI is happy we can land.

**Comment by sigito** [comment]
Done.

**Review by jmdetloff — APPROVED** [review]

**Pull request merged**